index/postings.go

```diff
@@ -268,17 +268,23 @@ type Postings interface {
 
 	// Err returns the last error of the iterator.
 	Err() error
+
+	// Reset resets the postings using the given data.
+	// It is similar to creating a new Postings using that data.
+	// If Reset() was successful, it returns true.
+	Reset(data interface{}) bool
```

Comment:
After another look, I think `Reset` is not required in that interface, because all the places where I use, I am first dereferencing and then using it. I added another commit which removes this methods and keeps it only for `ListPostings` and `BigEndianPostings`.

> `Clear(...)` maybe?

I remember seeing `Reset` in a lot of other places for similar use (start the object fresh with new data). `Clear` sounds more like clearing the object without setting the new data.

Reply:
> Clear sounds more like clearing the object without setting the new data.

Reset suggests the same thing to me. What about `Overwrite`?

Reply:
I would like to keep it `Reset` to be consistent.